bitcathedrals/buffer-ring
# Miscellaneous modifications

### Summary of changes

- Buffers can now be part of many rings, so the focus of the interfaces has moved
  from being buffer-centric to being ring-centric
- Lots of refactoring and fixes
- Tests for all interfaces
- Keep rings consistent with buffer state and usage via hooks and advice
- "Surface" ring in all member buffers whenever it becomes current, so that navigating
  the ring from any buffer traverses the correct "current" ring
- Rename functions for consistency and convention
- Ability to see current ring name
- Add buffer to default ring if none specified
- Use a buffer ring registry instead of buffer locals for buffer->rings lookup
  (Note: I'm not convinced this registry approach is better -- depending on how you feel,
  maybe we could revert to buffer locals down the line if we think that's better!)
- Use lexical binding
- Build scripts and dev workflow
- Lint and checkdoc
